src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -105,21 +138,54 @@ public BlockLoc getBlock(int x, int y, int z) {
     }
 
     @Override
-    public Biome getBiome(Vector3d position) {
-        throw new UnsupportedOperationException();
+    public BiomeType getBiome(Vector3i position) {
+        return (BiomeType) getBiomeGenForCoords(new BlockPos(position.getX(), position.getY(), position.getZ()));
     }
 
     @Override
+    public void setBiome(Vector3i position, BiomeType biome) {
+        net.minecraft.world.chunk.Chunk chunk = getChunkFromBlockCoords(new BlockPos(position.getX(), position.getY(), position.getZ()));
+        byte[] biomeArray = chunk.getBiomeArray();
+        // Taken from Chunk#getBiome
+        int i = position.getX() & 15;
+        int j = position.getZ() & 15;
+        biomeArray[j << 4 | i] = (byte) (((BiomeGenBase) biome).biomeID & 255);
+        chunk.setBiomeArray(biomeArray);
+    }
+
+    @SuppressWarnings("unchecked")
     public Collection<Entity> getEntities() {
-        return new ArrayList<Entity>();
+        return (Collection<Entity>) (Object) loadedEntityList;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Collection<Entity> getEntities(Predicate<Entity> filter) {
+        return (Collection<Entity>) (Object) func_175644_a(net.minecraft.entity.Entity.class,
+                (Predicate<net.minecraft.entity.Entity>) (Object) filter);
     }
 
     @Override
     public Optional<Entity> createEntity(EntityType type, Vector3d position) {
+        try {
+            net.minecraft.entity.Entity entity =
+                    (net.minecraft.entity.Entity) type.getEntityClass().getConstructor(net.minecraft.world.World.class).newInstance(this);
+            entity.setPosition(position.getX(), position.getY(), position.getZ());
+            if (spawnEntityInWorld(entity)) {
+                return Optional.of((Entity) entity);
+            }
+        } catch (Exception ignored) {
+            // Entity constructor does not have single World parameter, too bad
+        }
         return Optional.absent();
     }
```

Comment:
I suppose it might be, but since we already know the class (`type.getEntityClass()`) it's probably quicker to use that than to use a map lookup